### PR TITLE
holders tab changes

### LINKED ISSUE #5696: Holders tab changes
imageimage
- we are changing it a bit so its bit useful and not confusing 

new category: 
- longest hold 
- holding for on the detail of the nfts

updated in figma: collection detail handoff: https://www.figma.com/file/7BWdGiowrvTucpSFLE0kRX/Collection-detail-handoff?node-id=0%3A1&t=eGXVPezmA8IFsHk7-1

### PR DESCRIPTION
**Pull request by iMac7**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5696 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=14dLrAvjDw7cFLaxM7VWR6YtHd7bGZY2QpzXDhAJJ4yv6oGw&usdamount=undefined)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screenshot_20240305_152848

### CODE REVIEW

**Comment by exezbcz** [comment]
i think its missing the "bought xx days ago" 


image


As the issue is 
image

**Comment by exezbcz** [comment]
and also i think the holding for should be "longest hold" 

so, longest time period out of the all items

lets say you hold items for: 
10, 12, 20, 40 days 

longest hold = 40 

it is on the items you own, not all time

File: composables/collectionActivity/helpers.ts [hunk 1]
```diff
@@ -92,7 +93,19 @@ const updateOwnerWithNewNft = ({
...
       : owner.lastActivityTimestamp
+
   owner.nfts = [...owner.nfts, nft]
+
+  let longestHoldTimestamp = lastestTimeStamp
+
+  owner.nfts.forEach((nft) => {
+    longestHoldTimestamp = Math.max(
+      longestHoldTimestamp,
+      new Date(nft.updatedAt).getTime(),
```
Comment (Jarsen136) [inline composables/collectionActivity/helpers.ts:104]: Is this time correct? If user relists the nft with a new price, the `updatedAt` may change. It does not represent holding time.

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

File: composables/collectionActivity/helpers.ts (hunk 1 above)
Comment (Jarsen136) [reply composables/collectionActivity/helpers.ts:104]: Thanks for showing an example. You're right.

File: components/collection/activity/ownerInsightsTabs/HolderTab.vue [hunk 2]
```diff
@@ -80,6 +82,7 @@ import { Owners } from '@/composables/collectionActivity/types'
 import { NeoIcon } from '@kodadot1/brick'
 
 import NFTsDetaislDropdown from './NFTsDetaislDropdown.vue'
+import { formatSecondsToDuration } from '~/utils/format/time'
```
Comment (Jarsen136) [inline components/collection/activity/ownerInsightsTabs/HolderTab.vue:85]:
```suggestion
import { formatSecondsToDuration } from '@/utils/format/time'
```

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
let's see if i understood correctly: 
Take this [User](https://deploy-preview-9633--koda-canary.netlify.app/ahp/u/13zFVp9pdsJCJPeVBBRiSAkhBgZ4qENVEZBvfbs9Ft4rHVM8?collections=82) for example, he owns [nine Alchemy NFTs](https://github.com/kodadot/nft-gallery/assets/36627808/ac0055c4-e74b-451a-9f3e-dd46dabec68f) and in the list inside collection i can see that all of them but one is being held for 22 days

image

image

wouldn't it be the longest hold 1 month? since he still owns the nfts? 

collection for analysis: https://deploy-preview-9633--koda-canary.netlify.app/ahp/collection/82/activity

@exezbcz wdyt?

**Comment by exezbcz** [comment]
one last thing

hover over items: 

deploy:
image



canary: 
image

**Comment by exezbcz** [comment]
@iMac7 thank you 


@preschian could you please review? 👀

**Review by preschian — APPROVED** [review]
code lgtm

**Review by exezbcz — APPROVED** [review]

**Comment by prury** [comment]
by looking at the holders i still see incorrect info about the longest holder
go to: https://deploy-preview-9633--koda-canary.netlify.app/ahp/collection/82/activity
search for vikiival:
image

he is holding this nft for more than 1 month: https://deploy-preview-9633--koda-canary.netlify.app/ahp/gallery/82-198

**Comment by yangwao** [comment]
I see it's thing in interpretation until it hits 2 months, showing 1 month, do I get it correctly?

**Comment by iMac7** [comment]
> I see it's thing in interpretation until it hits 2 months, showing 1 month, do I get it correctly?

Yes

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 10.66 USD/DOT ~ 4.69 $DOT
🧗 [14dLrAvjDw7cFLaxM7VWR6YtHd7bGZY2QpzXDhAJJ4yv6oGw](https://polkadot.subscan.io/account/14dLrAvjDw7cFLaxM7VWR6YtHd7bGZY2QpzXDhAJJ4yv6oGw)
🔗 [0x265eca75c1d14e7bdb7665e5c440454613aa58721a12ae2735fa6bdf55ae101a](https://polkadot.subscan.io/extrinsic/0x265eca75c1d14e7bdb7665e5c440454613aa58721a12ae2735fa6bdf55ae101a)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**